core/trino-main/src/main/java/io/trino/operator/aggregation/AccumulatorCompiler.java

```diff
@@ -503,12 +503,39 @@ private static BytecodeBlock generateInputForLoop(
         Variable positionVariable = scope.declareVariable(int.class, "position");
         Variable rowsVariable = scope.declareVariable(int.class, "rows");
 
+        // prepare mayHaveNull condition
+        BytecodeExpression mayHaveNullCondition = BytecodeExpressions.constantFalse();
+        for (int parameterIndex = 0; parameterIndex < parameterVariables.size(); parameterIndex++) {
+            if (!argumentNullable.get(parameterIndex)) {
+                mayHaveNullCondition = BytecodeExpressions.or(mayHaveNullCondition, parameterVariables.get(parameterIndex).invoke("mayHaveNull", boolean.class));
+            }
+        }
+
         BytecodeBlock block = new BytecodeBlock()
                 .append(arguments)
                 .invokeVirtual(Page.class, "getPositionCount", int.class)
                 .putVariable(rowsVariable)
                 .initializeVariable(positionVariable);
 
+        ForLoop nullCheckLoop = generateInputLoopBody(true, scope, stateField, positionVariable, parameterVariables, lambdaProviderFields, inputFunction, callSiteBinder, grouped, argumentNullable, masksBlock, rowsVariable);
+        ForLoop noNullCheckLoop = generateInputLoopBody(false, scope, stateField, positionVariable, parameterVariables, lambdaProviderFields, inputFunction, callSiteBinder, grouped, argumentNullable, masksBlock, rowsVariable);
+
+        IfStatement mayHaveNullIf = new IfStatement("if(%s)", mayHaveNullCondition).condition(mayHaveNullCondition)
```

Comment:
Are there tests that cover this case (both edges)?

Reply:
Consider adding tests to `TestAccumulatorCompiler`